NEasyAuthMiddleware/ServiceBuilderExtensions.cs

```diff
@@ -12,6 +14,10 @@ public static class ServiceBuilderExtensions
     {
         public static AuthenticationBuilder AddEasyAuth(this AuthenticationBuilder builder, string authenticationScheme, Action<EasyAuthOptions> configure = null)
         {
+            if (!builder.Services.Any(s => typeof(IHttpContextAccessor).IsAssignableFrom(s.ServiceType)))
+            {
+                builder.Services.AddHttpContextAccessor();
+            }
```

Comment:
Why not use the `TryAddScoped` method?

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.tryaddscoped?view=net-8.0

Reply:
`services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();`

Reply:
I felt it would be better to try and use the "official" `AddHttpContextAccessor()` extension method, in the rare event Microsoft makes some update to how that functionality gets registered in the future.

But I have no strong opinion there and can update if desired!

Reply:
Actually, looking at the implementation of `AddHttpContextAccessor()`, it looks like MS already makes an effort to not double-register by using `TryAddSingleton()` themselves 😄 So looks like doing the check is unnecessary and we could just unconditionally call it!